pkg/controller/vsphere/reconciler.go

```diff
@@ -310,6 +321,24 @@ func (r *Reconciler) delete() error {
 	return fmt.Errorf("destroying vm in progress, reconciling")
 }
 
+// shouldWaitForNodeVolumes returns true if node status still have volumes attached
+// pod deletion and volume detach happen asynchronously, so pod could be deleted before volume detached from the node
+// this could cause issue for some storage provisioner, for example, vsphere-volume this is problematic
+// because if the node is deleted before detach success, then the underline VMDK will be deleted together with the Machine
+// so after node draining we need to check if all volumes are detached before deleting the node.
+func (r *Reconciler) shouldWaitForNodeVolumes(ctx context.Context, nodeName string, machineName string) (bool, error) {
```

Comment:
Perhaps naming this something like `nodeHasVolumesAttached` might make it more obvious what the bool return value here is? WDYT?